File: features/merchant/favorite/src/main/java/com/tokopedia/favorite/data/FavoriteDataRepository.java
package com.tokopedia.favorite.data;

import com.tokopedia.abstraction.common.utils.TKPDMapParam;
import com.tokopedia.favorite.domain.FavoriteRepository;
import com.tokopedia.favorite.domain.model.FavShop;
import com.tokopedia.favorite.domain.model.FavoriteShop;
import com.tokopedia.favorite.domain.model.TopAdsShop;

import java.util.HashMap;

import rx.Observable;

/**
 * @author Kulomady on 1/18/17.
 */

public class FavoriteDataRepository implements FavoriteRepository {
    private final FavoriteFactory favoriteFactory;

    public FavoriteDataRepository(FavoriteFactory favoriteFactory) {
        this.favoriteFactory = favoriteFactory;
    }

    @Override
    public Observable<FavoriteShop> getFirstPageFavoriteShop(HashMap<String, String> params) {
        return favoriteFactory.getFavoriteShopFirstPage(params);
    }

    @Override
    public Observable<FavoriteShop> getFavoriteShop(
            HashMap<String, String> param) {

        return favoriteFactory.getFavoriteShop(param);
    }

    @Override
    public Observable<TopAdsShop> getFreshTopAdsShop(HashMap<String, Object> params) {
        return favoriteFactory.getFreshTopAdsShop(params);
    }

    @Override
    public Observable<TopAdsShop> getTopAdsShop(HashMap<String, Object> params) {
        return favoriteFactory.getTopAdsShop(params);
    }
}

File: features/merchant/favorite/src/main/java/com/tokopedia/favorite/data/FavoriteDataRepository.kt
package com.tokopedia.favorite.data

import com.tokopedia.favorite.domain.FavoriteRepository
import com.tokopedia.favorite.domain.model.FavoriteShop
import com.tokopedia.favorite.domain.model.TopAdsShop
import java.util.*

/**
 * @author Kulomady on 1/18/17.
 */
class FavoriteDataRepository(private val favoriteFactory: FavoriteFactory) : FavoriteRepository {

    override suspend fun suspendGetFirstPageFavoriteShop(params: HashMap<String, String>): FavoriteShop {
        return favoriteFactory.suspendGetFavoriteShop(params)
    }

    override suspend fun suspendGetFavoriteShop(param: HashMap<String, String>): FavoriteShop {
        return favoriteFactory.suspendGetFavoriteShop(param)
    }

    override suspend fun suspendGetFreshTopAdsShop(params: HashMap<String, Any>): TopAdsShop {
        return favoriteFactory.suspendGetFreshTopAdsShop(params)
    }

    override suspend fun suspendGetTopAdsShop(params: HashMap<String, Any>): TopAdsShop {
        return favoriteFactory.suspendGetTopAdsShop(params)
    }

}
